spacy/pipeline/span_finder.py

```diff
@@ -0,0 +1,386 @@
+
+    model (Model[List[Doc], Floats2d]): A model instance that
+        is given a list of documents and predicts a probability for each token.
+    threshold (float): Minimum probability to consider a prediction positive.
+    predicted_key (str): Name of the span group the predicted spans are saved
+        to
+    training_key (str): Name of the span group the training spans are read
+        from
+    max_length (int): Max length of the produced spans (no max limitation when
+        set to 0)
+    min_length (int): Min length of the produced spans (no min limitation when
+        set to 0)
+    """
+    return SpanFinder(
+        nlp,
+        model=model,
+        threshold=threshold,
+        name=name,
+        scorer=scorer,
+        max_length=max_length,
+        min_length=min_length,
+        predicted_key=predicted_key,
+        training_key=training_key,
+    )
+
+
+@registry.scorers("spacy.span_finder_scorer.v1")
+def make_span_finder_scorer(
+    predicted_key: str = DEFAULT_PREDICTED_KEY,
+    training_key: str = DEFAULT_SPANS_KEY,
+):
+    return partial(
+        span_finder_score, predicted_key=predicted_key, training_key=training_key
+    )
+
+
+def span_finder_score(
+    examples: Iterable[Example],
+    *,
+    predicted_key: str = DEFAULT_PREDICTED_KEY,
+    training_key: str = DEFAULT_SPANS_KEY,
+    **kwargs,
+) -> Dict[str, Any]:
+    kwargs = dict(kwargs)
+    attr_prefix = "span_finder_"
+    kwargs.setdefault("attr", f"{attr_prefix}{predicted_key}")
+    kwargs.setdefault("allow_overlap", True)
+    kwargs.setdefault(
+        "getter", lambda doc, key: doc.spans.get(key[len(attr_prefix) :], [])
+    )
+    kwargs.setdefault("labeled", False)
+    kwargs.setdefault("has_annotation", lambda doc: predicted_key in doc.spans)
+    # score_spans can only score spans with the same key in both the reference
+    # and predicted docs, so temporarily copy the reference spans from the
+    # reference key to the candidates key in the reference docs, restoring the
+    # original span groups afterwards
+    orig_span_groups = []
+    for eg in examples:
+        orig_span_groups.append(eg.reference.spans.get(predicted_key))
+        if training_key in eg.reference.spans:
+            eg.reference.spans[predicted_key] = eg.reference.spans[training_key]
+    scores = Scorer.score_spans(examples, **kwargs)
+    for orig_span_group, eg in zip(orig_span_groups, examples):
+        if orig_span_group is not None:
+            eg.reference.spans[predicted_key] = orig_span_group
+    return scores
+
+
+class SpanFinder(TrainablePipe):
+    """Pipeline that learns span boundaries"""
+
+    def __init__(
+        self,
+        nlp: Language,
+        model: Model[Iterable[Doc], Floats2d],
+        name: str = "span_finder",
+        *,
+        threshold: float = 0.5,
+        max_length: int = 0,
+        min_length: int = 0,
+        # XXX I think this is weird and should be just None like in
```

Comment:
Many of the existing components have defaults here so that the behavior of `nlp.add_pipe("name")` didn't change as scorers were added as separate registered functions. It's possible we could do this different for a new component?

Reply:
Sorry. The default `max_positive` is `None` in the `SpanFinder` and I think that makes more sense for `max_length` as well because `max_length=0` is really weird semantically I think that the "unlimited" is 0.